src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/GeneratedRegexAttribute.cs

```diff
@@ -8,7 +8,17 @@
 namespace System.Text.RegularExpressions;
 
 /// <summary>Instructs the System.Text.RegularExpressions source generator to generate an implementation of the specified regular expression.</summary>
-/// <remarks>The generator associated with this attribute only supports C#.  It only supplies an implementation when applied to static, partial, parameterless, non-generic methods that are typed to return <see cref="Regex"/>.</remarks>
+/// <remarks>
+/// The generator associated with this attribute only supports C#.  It only supplies an implementation when applied to static, partial, parameterless, non-generic methods that
+/// are typed to return <see cref="Regex"/>.
+///
+/// When the regex supports case-insensitive matches (either by passing <see cref="RegexOptions.IgnoreCase"/> or using the inline `(?i)` switch in the pattern) the regex
+/// engines will use an internal casing table to transform the passed in pattern into a case-sensitive equivalent one. For example, given the pattern `abc` our engines
+/// will transform it to `[Aa][Bb][Cc]`. The equivalences found on this internal casing table can change over time, for example in the case new characters are added to
+/// a new version of Unicode. When using the source generator's engine, this transformation happens at compile time, which means that the casing table used to find the
+/// equivalences will be depending on the target framework at compile time. This differs from the rest of our engines, which will make the transformation at runtime, meaning
+/// it will always use the most current casing table for the current runtime. This discrepancy is explained further here: https://github.com/dotnet/runtime/issues/70214
```

Comment:
do we usually refer to github issues in our docs? I suggest either removing the last sentence completely or have a public doc describing the discrepancy.

Reply:
I was also on the fence about this. I'll remove it from the description.